File: shopping-list.py
```
def shopping_list():
    prices = list(map(float, input().split()))
    discounts = list(map(float, input().split()))

    n = len(prices)  
    total_price = sum(price * (1 - discount / 100) for price, discount in zip(prices, discounts))

    return round(total_price, 1)


print(shopping_list())

File: shopping-list.cpp
#include <vector>
using namespace std;

double shoppingList(double price[], int n, double discount[]) {
    double total = 0;
    for (int i = 0; i < n; i++) {
        double p = price[i];
        double d = p * discount[i] / 100;
        total += p - d;
    }
    return total;
}

int main() {
    int N;
    cin >> N;
    vector<double> prices(N);
    for (auto& p : prices) {
        cin >> p;
    }
    vector<double> discounts(N);
    for (auto& d : discounts) {
        cin >> d;
    }
    cout << fixed << setprecision(1) << shoppingList(&prices[0], N, &discounts[0]) << endl;
    return 0;
}